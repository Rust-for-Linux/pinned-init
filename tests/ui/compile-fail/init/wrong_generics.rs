use pin_init::*;

struct Foo<T> {
    value: T,
}
fn main() {
    let _ = init!(Foo<()> {
        value <- (),
    });
}
