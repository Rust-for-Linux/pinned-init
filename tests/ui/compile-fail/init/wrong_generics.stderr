error: comparison operators cannot be chained
 --> tests/ui/compile-fail/init/wrong_generics.rs:7:22
  |
7 |     let _ = init!(Foo<()> {
  |                      ^  ^
  |
help: use `::<...>` instead of `<...>` to specify lifetime, type, or const arguments
  |
7 |     let _ = init!(Foo::<()> {
  |                      ++

error: comparison operators cannot be chained
 --> tests/ui/compile-fail/init/wrong_generics.rs:7:22
  |
7 |     let _ = init!(Foo<()> {
  |                      ^  ^
  |
  = help: use `::<...>` instead of `<...>` to specify lifetime, type, or const arguments
  = help: or use `(...)` if you meant to specify fn arguments
